packages/core/src/components/Tabs/Tabs.jsx

```diff
@@ -134,6 +175,75 @@ export class Tabs extends React.PureComponent {
     }
   }
 
+  /**
+   * Build tabs array to handle arrow keypress events accurately
+   */
+  tabsArrayLen() {
+    return React.Children.toArray(this.props.children)
+      .filter(isTab);
+  }
+
+  /**
+   * Find the current tab index on left or right arrow keypress
+   * @param {Array} React Tab components
+   * @param {String} ID prop of currently selected Tab
+   * @return {Number} Returns index of currently selected array item
+   */
+  tabFindIndex(tabsArr, findId) {
+    const index = tabsArr.findIndex(elem => elem.props.id === findId);
+
+    return index;
+  }
+
+  /**
+   * Handle the left arrow keydown events
+   * @param {Object} Native event object, used to listen for left arrow keycode
+   * @param {String} ID prop of currently selected Tab
+   * 
+   * The index returned from tabFindIndex is used in a reduce by 1 logic
+   * tree to ensure the right tab[index] ID is targeted for updated state.
+   * 
+   * The replaceState function did not like the href argument passed into 
+   * handleTabClick, so passing an interpolated hashbang plus selected Tab
+   * string instead. 
+   */
+  tabSwitchLeft(evt, curId) {
+    const tabs = this.tabsArrayLen();
+    const index = this.tabFindIndex(tabs, curId);
+
+    if (index === 0) {
+      const target = tabs[tabs.length - 1].props.id;
+      this.setState({ selectedId: target });
+      this.replaceState(`#${target}`);
```

Comment:
We should use the Tab's `href` prop, which is returned as the forth argument of `onKeyDown`. It's not guaranteed to be a hash, so doing it this way won't always work. This also applies to the other lines where this is being done.

Reply:
@sawyerh I wondered about using the href attribute, but it was returning the wrong value. It's possible because I'm passing in the tab, not the tabPanel for ID that I'm getting the wrong value. I'll continue to review and clean up accordingly.